internal/configs/configurator.go

```diff
@@ -1132,20 +1125,23 @@ func (cnf *Configurator) UpdateConfig(cfgParams *ConfigParams, resources Extende
 		allWarnings.Add(warnings)
 	}
 
-	// we don't need to regenerate config for TransportServers, because:
-	// (1) Changes to the ConfigMap don't affect TransportServer configs directly
-	// (2) addOrUpdateTransportServer doesn't return any warnings that we need to propagate to the caller.
-	// if (1) and (2) is no longer the case, we need to generate the config for TransportServers
+	for _, tsEx := range resources.TransportServerExes {
+		warnings, err := cnf.addOrUpdateTransportServer(tsEx)
+		if err != nil {
+			return allWarnings, err
```

Comment:
variable ```warnings``` (line 1129) is irrelevant when ```err``` is returned, therefore we return ```allWarnings``` and the ```err``` on the line 1131. Please note that although it works it is not entirely correct. The outer func at this point should return ```nil``` and the error. Currently, the implementation pattern (applied in few more places in this package) can lead into thinking that returned ```warnings``` ALONG with not nil ```err``` can be used, which is not correct.

In short, lines 1128, 1132 should read:
```go
for _, tsEx := range resources.TransportServerExes {
    warnings, err := cnf.addOrUpdateTransportServer(tsEx)
    if err != nil {
        return nil, err
    }
    allWarnings.Add(warnings)
```
If ```cnf.addOrUpdateTransportServer(tsEx)``` returns warnings they are added to allWarnings and returned by the outer function at the end:
```go
return allWarnings, nil
```

Correcting of the returning values will be done in a separate PR as it will require small refactoring of functions/methods and corresponding unit tests.